pkg/pmem-device-manager/pmd-util.go

```diff
@@ -81,3 +84,29 @@ func WaitDeviceAppears(dev PmemDeviceInfo) error {
 	}
 	return fmt.Errorf("device %s did not appear after multiple retries", dev.Path)
 }
+
+func AdjustSizeLVM(size uint64) (uint64, uint64) {
+	// adjust up to next 4MB boundary as LVM would do it anyway
+	size /= lvmAlign
+	size += 1
+	size *= lvmAlign
+	return size, lvmAlign
+}
+
+func AdjustSizeNdctl(size uint64) (uint64, uint64) {
+	if size <= 0 {
+		// Allocating volumes of zero size isn't supported.
+		// We use some arbitrary small minimum size instead.
+		// It will get rounded up by libndctl to meet the alignment.
+		size = 1
+	}
+	// libndctl needs to store meta data and will use some of the allocated
+	// space for that (https://github.com/pmem/ndctl/issues/79).
+	// We don't know exactly how much space that is, just
+	// that it should be a small amount. But because libndctl
+	// rounds up to the alignment, in practice that means we need
+	// to request `align` additional bytes.
+	size += ndctlAlign
```

Comment:
It shouldn't be that adding extra 1GB to size only in case when size < 2GB ?

Reply:
IIRC, this was not size dependent. We generally always get one align-step less than what we asked. This is relatively easy to verify with cmdline ndctl, creating namespaces and checking what is created.

Reply:
Ya got it. Thanks.

Reply:
Then this issue exists with even default(2M) alignment, so I think we better handle this ndctl layer.

Reply:
If we move handling of size compensation out of here into ndctl layer, then even less code remains here, and also less gets covered by unit testing that I planned to add here.
The whole idea of having a separate function for align and size calc gets weaker. It is already quite weak to start with, because major factor affecting final values: InterleaveWidth is inside ndctl and not usable during unit test time as it's HW-specific.
So I have also thinking, should we give up idea what is in this PR altogether.
Comments cleanup seems a good part here, though, as some existing comments are out of date.